module/VuFind/src/VuFind/Http/GuzzleService.php

```diff
@@ -98,11 +110,94 @@ public function __construct(array $config)
      *
      * @return \GuzzleHttp\ClientInterface
      */
-    public function createClient(?string $url = null, ?float $timeout = null): \GuzzleHttp\ClientInterface
+    public function createGuzzleClient(?string $url = null, ?float $timeout = null): \GuzzleHttp\ClientInterface
     {
         return new \GuzzleHttp\Client($this->getGuzzleConfig($url, $timeout));
     }
 
+    /**
+     * Perform a GET request.
+     *
+     * @param string $url     Request URL
+     * @param array  $params  Request parameters (query string)
+     * @param float  $timeout Request timeout in seconds
+     * @param array  $headers Request HTTP headers
+     *
+     * @return ResponseInterface
+     */
+    public function get(
+        string $url,
+        array $params = [],
+        ?float $timeout = null,
+        array $headers = []
+    ): ResponseInterface {
+        if ($params) {
+            $query = $this->createQueryString($params);
+            if (str_contains($url, '?')) {
+                $url .= '&' . $query;
+            } else {
+                $url .= '?' . $query;
+            }
+        }
+        $client = $this->createGuzzleClient($url, $timeout);
+        $options = [];
+
+        if ($headers) {
+            $options['headers'] = $headers;
+        }
+
+        return $client->request('GET', $url, $options);
+    }
+
+    /**
+     * Perform a POST request.
+     *
+     * @param string $url     Request URL
+     * @param string $body    Request body document
+     * @param string $type    Request body content type
+     * @param float  $timeout Request timeout in seconds
+     * @param array  $headers Request HTTP headers
+     *
+     * @return ResponseInterface
+     */
+    public function post(
+        string $url,
+        $body = null,
```

Comment:
Minor suggested type adjustments:

```suggestion
     * @param string  $url     Request URL
     * @param ?string $body    Request body document
     * @param string  $type    Request body content type
     * @param float   $timeout Request timeout in seconds
     * @param array   $headers Request HTTP headers
     *
     * @return ResponseInterface
     */
    public function post(
        string $url,
        ?string $body = null,
```